### DISCUSSION TITLE
listen() function returns -1, however errno is set to 0

### DISCUSSION THREAD

**Discussion by seogyeongjeong** [ANLAB-KAIST/KENSv3#43] [Project Q&A 1. Application Layer (Socket)] [answered]
I tried listen() function this way, however my echoserverfunction seems to have trouble in listen()
One thing I cannot understand is, errno is set to 0.
According to the manpage [listen(2)](https://man7.org/linux/man-pages/man2/listen.2.html), if listen() returns -1 then errno will be set to indicate the error.
I have tried a lot of times to get over this. Can I get some hint?
![image](https://user-images.githubusercontent.com/79819045/160253954-7132a7ae-ae6e-4a97-9b3e-e6f9b1704d24.png)
![image](https://user-images.githubusercontent.com/79819045/160253965-e43d4abc-95df-4608-b6c4-7881a081e74e.png)

Would this suggest that I might have converted appropriate IP address in a wrong way?

**Comment by Pusnow** [comment, marked as answer]
KENS does not support errno. Try non-kens version to debug this.

**Reply by seogyeongjeong** [reply]
Uhm, I'm so sorry but I tried non-kens version by 
`./app/echo/echo-non-kens s 127.0.0.1 9000 hello`
And it shows the same result. (listen() function returns -1 and errno is set to 0).
:-(
